lib/engine/round/base.rb

```diff
@@ -217,6 +217,11 @@ def can_gain?(share, entity)
         corporation.holding_ok?(entity, share.percent) &&
         (!corporation.counts_for_limit || entity.num_certs < @game.cert_limit)
       end
+
+      def did_sell?(corporation, entity)
+        is_a?(Engine::Round::Stock) &&
```

Comment:
we should not import round stock in here as it might induce a circular dependency

intsead this should return false, and you should override did_sell? in round/stock